src/main/java/de/triology/cas/ldap/ConnectionAwareAuthenticator.java

```diff
@@ -45,6 +47,17 @@ public ConnectionAwareAuthenticator(
         this.useUserConnectionToFetchAttributes = useUserConnectionToFetchAttributes;
     }
     
+    public void someBuggyMethod() {
+        LdapEntry entry = null;       
+        if (entry == null) {
+            System.out.println(entry.getAttribute());
+        }
+        
+        if (entry != null) {
+            entry = null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "entry". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1854)